services/core/shared/core/transactionStatistics.js

```diff
@@ -260,10 +261,41 @@ const init = async historyLengthDays => {
 
 const updateTodayStats = async () => fetchTransactionsForPastNDays(1, true);
 
+const updateTransactionStatistics = async historyLengthDays => {
+	const dateTo = moment().utc().endOf('day').subtract(0, 'day');
+	const dateFrom = moment(dateTo).startOf('day').subtract(historyLengthDays, 'day');
+
+	const params = {
+		dateFormat: 'YYYY-MM-DD',
+		dateTo,
+		dateFrom,
+	};
+
+	const distributionByType = await getDistributionByType(params);
+
+	const verifyStatistics = await BluebirdPromise.map(
+		Object.keys(distributionByType),
+		async type => {
+			const txCount = (await getTransactions({
+				moduleAssetId: type,
+				fromTimestamp: (moment.unix(dateFrom).unix()) / 1000,
+				toTimestamp: (moment.unix(dateTo).unix()) / 1000,
+			})).meta.total;
+			const result = txCount === distributionByType[type];
+			return result;
+		},
+		{ concurrency: Object.keys(distributionByType).length },
+	);
+
+	const isStatisticsBuilt = !verifyStatistics.some(acc => acc === false);
+	if (!isStatisticsBuilt) await fetchTransactionsForPastNDays(historyLengthDays, true);
```

Comment:
Kindly consider simplifying the code to something similar to:

```suggestion
	const isStatsProperlyBuilt = verifyStatistics.every(isVerified => isVerified);
	if (!isStatsProperlyBuilt) await fetchTransactionsForPastNDays(historyLengthDays, true);
```